modules/gdscript/gdscript_editor.cpp

```diff
@@ -2117,6 +2117,7 @@ static bool _guess_identifier_type(GDScriptParser::CompletionContext &p_context,
 	int last_assign_line = -1;
 	const GDScriptParser::ExpressionNode *last_assigned_expression = nullptr;
 	GDScriptCompletionIdentifier id_type;
+	Variant id_value;
```

Comment:
`id_type` is now a `GDScriptCompletionIdentifier` so we can store the value in `id_type.value` directly. It might be necessary to reset the value when reassigning the type.